src/components/LinkWithSearchParams.tsx

```diff
@@ -0,0 +1,9 @@
+import { Link, useSearchParams, type LinkProps } from "wouter"
+
+export function LinkWithSearchParams({ to, href, ...props }: LinkProps) {
```

Comment:
I've considered and tried using the hook approach by listening to `useLocation` and `useSearchParams` and programmatically set the param back. This wasn't great because the links are still rendered without the parameter, so when users open such links in new tabs, they won't have the search parameters. This fixes the problem by rendering the correct link in the first place.